### DISCUSSION TITLE
How to collapse multiple cursors?

### DISCUSSION THREAD

**Discussion by mankinskin** [helix-editor/helix#503] [Q&A] [answered]
Hi, is there a way to collapse multiple cursors after selecting multiple pieces using `s` in a selection?
![Peek 2021-07-25 13-53](https://user-images.githubusercontent.com/20745737/126898036-f7631f6a-6b83-490e-9c0b-ba9cdbd6b33f.gif)

what I did here was `v`, move down, `s`, type "response", Enter, `ms)`, `u`, move around.

Is there a key to collapse the multiple cursors into one, like the top one? I know of `;` but that doesn't do it.

**Comment by sudormrfbin** [comment, marked as answer]
`,` (comma) should do the trick. Versions prior to and including `0.4.1` use `<space> <space>` instead.

**Comment by pickfire** [comment]
I think we should have a tutorial like vimtutor for these additional functionalities.

**Comment by mankinskin** [comment]
@sudormrfbin that did it, thanks

@pickfire I don't think that is needed, I never really used the vimtutor either. I think a complete documentation and something like an extensive FAQ you can search would help the most.

**Reply by matu3ba** [reply]
I disagree insofar as having related things grouped together is not a FAQ structure.

**Reply by mankinskin** [reply]
> I disagree insofar as having related things grouped together is not a FAQ structure.

but what I am talking about is an FAQ, i.e. "How do I collapse multiple cursors? - `<space>` `<space>`"

**Comment by Omnikar** [comment]
TO OTHER USERS READING THIS DISCUSSION: `<space><space>` is no longer the mapping, it is now `,`.

**Reply by archseer** [reply]
(on master and from next release onward)

**Comment by qstommyshu** [comment]
Wouldn't ";" be more intuitive for collapsing multi cursors, as helix already have ";" to collapse selection? ";" can be a standard collapsing key press for helix?

**Reply by k3d3** [reply]
Agreed on this; I kept using semicolon to collapse multiple selections, and I was confused why it didn't work in this scenario but did in basically every other.

I suppose in hindsight it's a different operation (semicolon to remove the highlight, comma to un-highlight everything except the current one), but I definitely think it's best to keep the comma the same, but have the semicolon *also* collapse multiple selections.

**Comment by ghost** [comment]
So I was looking for how to close multiple cursors after using `c` and ended up discovering how powerful the selection feature is in this editor!

**Comment by gokayburuc** [comment]
press <kbd> Space + ? </kbd> and write `collapse_selection` you can find useful keybindings there.

**Reply by medanimohamedakram** [reply]
The `collapse_selection` would collapse any cursor selection to a single character cursor selection, not the number of cursor in the editor.
There should be something like `collapse_cursor` to  make things more obvious.
